### ISSUE TITLE
Add privacy info file to Xcode project

### ISSUE THREAD

**Issue by joshuatbrown** [planetary-social/nos#960]
As [this Mastodon thread](https://mastodon.social/@chockenberry/112095424613859371) explains, we'll need to add a privacy info file to our Xcode project to meet with Apple's updated requirements. We'll need to add `PrivacyInfo.xcprivacy` with some information about how we're using `UserDefaults`.

See [this Slack post](https://planetary-app.slack.com/archives/C04PLR5QFQQ/p1711074563930519) for warnings from Apple.

**Comment by joshuatbrown** [comment]
Just heard about this tool that may help: https://github.com/Wooder/ios_17_required_reason_api_scanner

**Comment by mplorentz** [comment]
Looks like we missed one here. I just got this email from Apple regarding 0.1.10 (238) production build:
```
ITMS-91053: Missing API declaration - Your app’s code in the “Nos” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryFileTimestamp. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.Apple Developer Relations
```

**Comment by joshuatbrown** [comment]
When I run the binary scanner from [this repo](https://github.com/Wooder/ios_17_required_reason_api_scanner), I get the following output:

```
Analyzing binaries: ./Nos Dev.app/Nos Dev ./Sentry.framework/Sentry ./Nos Dev.app/Frameworks/Sentry.framework/Sentry
---
Used symbols in binary ./Nos Dev.app/Nos Dev: NSFileCreationDate, NSFileModificationDate, NSURLContentModificationDateKey, NSURLCreationDateKey, NSUserDefaults, stat
Used symbols in binary ./Sentry.framework/Sentry: NSFileCreationDate, stat
```

This seems to indicate that Sentry is the culprit here. I've double-checked this by running the text scanner from the same repo, and it indicates that our source code only uses UserDefaults, as far as privacy requirements go.

I've checked the Sentry CHANGELOG, and it seems like they added the PrivacyInfo in 8.21.0. We're using 8.23.0, so I'm not sure why we'd still be seeing that message. Perhaps Sentry.framework doesn't include the PrivacyInfo file?

As a workaround, we can probably just add the necessary entry to _our_ `PrivacyInfo.xcprivacy` but I'd rather have Sentry include it in theirs. This would also help others who use Sentry and are dealing with the same issue.

**Comment by joshuatbrown** [comment]
I've reported this as [a bug in Sentry](https://github.com/getsentry/sentry-cocoa/issues/3853). Hopefully we get a response from them soon.

**Comment by joshuatbrown** [comment]
Well, Sentry [responded](https://github.com/getsentry/sentry-cocoa/issues/3853#issuecomment-2056742100) by saying their `PrivacyInfo.xcprivacy` file is included in the pre-compiled framework they distribute. I checked the framework and this seems correct to me -- I was able to find it in the framework:

<img width="493" alt="Screenshot 2024-04-15 at 10 55 25 AM" src="https://github.com/planetary-social/nos/assets/59564/789c380c-33d4-49a1-acf0-b64aea4f3bba">

I checked their privacy file and it does contain the expected `NSPrivacyAccessedAPICategoryFileTimestamp`, so now I'm stumped. Here's what I think we know:

1. We don't use any file timestamp APIs (based on both the binary _and_ text scanner results)
2. Sentry does (based on the binary scanner results)
3. Our other dependencies don't (based on the binary _and_ text scanner results)
4. Sentry includes the privacy file with the correct value for `NSPrivacyAccessedAPICategoryFileTimestamp`
5. Apple still warns us that we're "Missing API declaration" when we upload a build to TestFlight

Thoughts on what we're missing here, @mplorentz or @martindsq? We can still add the value to our `PrivacyInfo.xcprivacy`, but I don't see why we'd need to and I'd rather do that only if/when we understand _why_.

**Comment by joshuatbrown** [comment]
On further investigation, I think this may be due to CocoaLumberjack, which uses two File timestamp APIs: NSFileCreationDate and NSFileModificationDate. Its privacy file contains two reasons: C617.1 and 0A2A.1. The second reason _may_ be the reason we need to specify our own (but Apple's docs are insufficient). Here's what 0A2A.1 says:

> Declare this reason if your third-party SDK is providing a wrapper function around file timestamp API(s) for the app to use, and you only access the file timestamp APIs when the app calls your wrapper function.

So if we call something in CocoaLumberjack that uses a file timestamp API, does that mean we need to declare the `NSPrivacyAccessedAPICategoryFileTimestamp`? I don't know for sure (again, Apple's docs are woefully inadequate), but I'm willing to say that's why we still get warnings. I'll add that one to our PrivacyInfo.xcprivacy and we can be on our way.

**Comment by joshuatbrown** [comment]
@mplorentz Was 0.1.10 (238) built with Xcode 15.2? There's a [comment on my Sentry bug ticket](https://github.com/getsentry/sentry-cocoa/issues/3853#issuecomment-2058985010) that seems to indicate that might have been the problem. They say the issue is fixed in 15.3.

Of course, now that we added our own entry to our privacy file, this doesn't matter too much, but I'd still love to know.

**Comment by mplorentz** [comment]
@joshuatbrown unfortunately no, [here](https://github.com/planetary-social/nos/actions/runs/8663974068/job/23759295838) are the build logs, they show Xcode 15.3.
